imvertorlite.py

```diff
@@ -1,137 +1,189 @@
+                output = json.dumps(schema, indent=2)
+                with open(f"{schema['title'].lower()}.json", "w") as f:
+                    f.write(output)
+
+
+def process_enterprise_architect(file_name, class_name, encoding):
+    """
+    Process the enteprise architect format (XML)
+
+    @param soup: The base soup of the XML file to resolve references.
+    @param base: The base UML class (BeautifulSoup object).
+    @param attrs: A ResultSet (list of elements) containing all the individual properties.
+    """
+
+    with open(file_name, "r", encoding=encoding if encoding else "latin-1") as f:
+        data = f.read()
+
+    soup = BeautifulSoup(data, "lxml")
+
+    if class_name:
+        base_list = soup.find_all('element', {'xmi:type': 'uml:Class', 'name': class_name})
+    else:
+        # TODO: Filter out useful classes.
+        base_list = soup.find_all('element', {'xmi:type': 'uml:Class'})
+
+    for base in base_list:
+        attrs = base.select("attribute")
+        generate_schema(soup, base, attrs)
+
+
+def process_amsterdam_schema(file_name, encoding='utf-8'):
+    """
+    Process the Amsterdam Schema format (JSON)
+
+    @param file_name: The name of the JSON file.
+    @param encoding: The encoding of the given JSON file (Default: "utf-8").
+    """
+
+    with open(file_name, "r", encoding=encoding) as f:
+        obj = json.load(f)
+
+    schemas = obj['tables']
+    for schema in schemas:
+        schema['properties'] = schema.pop('schema')
+        schema['type'] = 'object'
+
+        for key in list(schema['properties'].keys()):
+            if key not in ['properties']:
+                schema[key] = schema['properties'].pop(key)
+
+        properties = schema['properties'].pop('properties')
+        for key, dic in properties.items():
+            if key not in ['schema']:
+                dic['$id'] = f'#/properties/{key}'
+                dic['title'] = key
+                schema['properties'][key] = dic
+
+        schema['required'].remove('schema')  # Remove 'schema' property
+
+        with open(f"{schema['title'].lower()}.json", "w") as f:
+            json.dump(schema, f, indent=2)
+
+
+@click.command()
+@click.option(
+    "file_name",
+    "--file",
+    "-f",
+    required=True,
+    help="Supported formats: Enterprise Architect XML, Amsterdam Schema JSON.",
+)
+@click.option(
+    "class_name", "--name", "-n", help="The UML class name of the object to export."
+)
+@click.option(
+    "encoding", "--encoding", "-e", help="Specify the encoding of the XML file (optional).",
+)
+def convert(file_name, class_name=None, encoding=None):
+    """
+    Exports one or more UML classes from an Enterprise Architect XML export to 
+    JSON Schema.
+    """
+
+    if file_name.endswith('.json'):
+        process_amsterdam_schema(file_name, encoding)
+    elif file_name.endswith('.xml'):
+        process_enterprise_architect(file_name, class_name, encoding)
+    else:

```

Comment:
I think I also prefer something more explicit like `--format=ea` or `--format=amsterdamschema` and if not provided THEN "autodetect" :)